src/OmniSharp/Api/v1/Navigation/OmnisharpController.GotoDefinition.cs

```diff
@@ -16,59 +18,16 @@ public partial class OmnisharpController
         [HttpPost("gotodefinition")]
         public async Task<GotoDefinitionResponse> GotoDefinition(GotoDefinitionRequest request)
         {
-            var quickFixes = new List<QuickFix>();
-
-            var document = _workspace.GetDocument(request.FileName);
-            var response = new GotoDefinitionResponse();
-            if (document != null)
+            var definitions = _workspace.PluginHost.GetExports<IGotoDefintion>();
+            foreach (var instance in definitions)
```

Comment:
This pattern is going to be the same for all request handlers. Would be nice to abstract it away somehow.